bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceDialog.java

```diff
@@ -1895,85 +1252,93 @@ protected int getDialogBoundsStrategy() {
 	}
 
 	/**
 	 * Stores its current configuration in the dialog store.
 	 */
 	private void writeConfiguration() {
-		IDialogSettings s= getDialogSettings();
-
-		s.put("wrap", fWrapInit); //$NON-NLS-1$
-		s.put("casesensitive", fCaseInit); //$NON-NLS-1$
-		s.put("wholeword", fWholeWordInit); //$NON-NLS-1$
-		s.put("incremental", fIncrementalInit); //$NON-NLS-1$
-		s.put("isRegEx", fIsRegExInit); //$NON-NLS-1$
-
-		List<String> history= getFindHistory();
-		String findString= getFindString();
-		if (!findString.isEmpty())
-			history.add(0, findString);
-		writeHistory(history, s, "findhistory"); //$NON-NLS-1$
-
-		history= getReplaceHistory();
-		String replaceString= getReplaceString();
-		if (!replaceString.isEmpty())
-			history.add(0, replaceString);
-		writeHistory(history, s, "replacehistory"); //$NON-NLS-1$
+		IDialogSettings s = getDialogSettings();
+
+		s.put("wrap", findReplaceLogic.isActive(SearchOptions.WRAP)); //$NON-NLS-1$
+		s.put("casesensitive", findReplaceLogic.isActive(SearchOptions.CASE_SENSITIVE)); //$NON-NLS-1$
+		s.put("wholeword", findReplaceLogic.isActive(SearchOptions.WHOLE_WORD)); //$NON-NLS-1$
+		s.put("incremental", findReplaceLogic.isActive(SearchOptions.INCREMENTAL)); //$NON-NLS-1$
+		s.put("isRegEx", findReplaceLogic.isActive(SearchOptions.REGEX)); //$NON-NLS-1$
+
+		String findString = getFindString();
+		findHistory.add(findString);
+
+		String replaceString = getReplaceString();
+		replaceHistory.add(replaceString);
+	}
+
+	private void activateInFindReplacerIf(SearchOptions option, boolean shouldActivate) {
+		if (shouldActivate) {
+			findReplaceLogic.activate(option);
+		} else {
+			findReplaceLogic.deactivate(option);
+		}
+	}
+
+	private void evaluateFindReplacerStatus() {
+		evaluateFindReplacerStatus(true);
 	}
 
 	/**
-	 * Writes the given history into the given dialog store.
+	 * Evaluate the status of the FindReplaceLogic object.
 	 *
-	 * @param history the history
-	 * @param settings the dialog settings
-	 * @param sectionName the section name
-	 * @since 3.2
+	 * @param beep Whether the evaluation should beep on some codes.
 	 */
-	private void writeHistory(List<String> history, IDialogSettings settings, String sectionName) {
-		int itemCount= history.size();
-		Set<String> distinctItems= new HashSet<>(itemCount);
-		for (int i= 0; i < itemCount; i++) {
-			String item= history.get(i);
-			if (distinctItems.contains(item)) {
-				history.remove(i--);
-				itemCount--;
-			} else {
-				distinctItems.add(item);
+	private void evaluateFindReplacerStatus(boolean beep) {
+		IStatus status = findReplaceLogic.getStatus();
+
+		String dialogMessage = status.accept(new FindReplaceLogicMessageGenerator());
+
+		fStatusLabel.setText(dialogMessage);
+
+		fStatusLabel.setForeground(null);
+		if (status instanceof FindStatus statusMessage) {
+			if (statusMessage.isErrorCode()) {
+				fStatusLabel.setForeground(JFaceColors.getErrorText(fStatusLabel.getDisplay()));
+			}
```

Comment:
Coloring also needs to be applied when `status` is an `InvalidRegExStatus`. Maybe `IStatus` should have an `isError()` method that can be used to retrieve whether the status is an error without the necessity to check for instances?

Reply:
I agree, this unknots the logic and helps by improving the meaning of the statuses.

Reply:
https://github.com/eclipse-platform/eclipse.platform.ui/commit/116106afacb33cc9225e7ec14c2f1c5c7f845ec1